homeassistant/components/lyric/__init__.py

```diff
@@ -0,0 +1,188 @@
+"""Support for Honeywell Lyric devices."""
+import asyncio
+import logging
+from typing import Any, Dict
+from datetime import timedelta
+
+import voluptuous as vol
+from lyric import Lyric
+
+from homeassistant.components.lyric import config_flow
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_TOKEN
+from homeassistant.helpers import config_validation as cv, config_entry_oauth2_flow
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.typing import ConfigType, HomeAssistantType
+
+from .const import (
+    DATA_LYRIC_CLIENT,
+    DOMAIN,
+    CONF_CLIENT_ID,
+    CONF_CLIENT_SECRET,
+    CONF_LYRIC_CONFIG_FILE,
+    SERVICE_HOLD_TIME,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+SCAN_INTERVAL = timedelta(seconds=10)
+
+CONFIG_SCHEMA = vol.Schema(
+    {
+        DOMAIN: vol.Schema(
+            {
+                vol.Required(CONF_CLIENT_ID): cv.string,
+                vol.Required(CONF_CLIENT_SECRET): cv.string,
+            }
+        )
+    },
+    extra=vol.ALLOW_EXTRA,
+)
+
+LYRIC_COMPONENTS = ["climate", "sensor"]
+
+
+async def async_setup(hass, config):
+    """Set up the Lyric component."""
+    hass.data[DOMAIN] = {}
+
+    if DOMAIN not in config:
+        return True
+
+    config_flow.LyricFlowHandler.async_register_implementation(
+        hass,
+        config_entry_oauth2_flow.LocalOAuth2Implementation(
+            hass,
+            DOMAIN,
+            config[DOMAIN][CONF_CLIENT_ID],
+            config[DOMAIN][CONF_CLIENT_SECRET],
+            "https://api.honeywell.com/oauth2/authorize",
+            "https://api.honeywell.com/oauth2/token",
+        ),
+    )
+
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistantType, entry: ConfigEntry) -> bool:
+    """Set up Lyric from a config entry."""
+    implementation = await config_entry_oauth2_flow.async_get_config_entry_implementation(
+        hass, entry
+    )
+
+    implementation.token = entry.data[CONF_TOKEN]
+    implementation.token_cache_file = hass.config.path(CONF_LYRIC_CONFIG_FILE)
```

Comment:
Why is there a token cache file and why don't we use `entry.data` to store things (which is the default for the local implementation)?

Reply:
Never mind this, let's get this merged and we can fix this in a future PR.